price-feeder/oracle/types/provider.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+const (
+	ProviderKraken   ProviderName = "kraken"
+	ProviderBinance  ProviderName = "binance"
+	ProviderOsmosis  ProviderName = "osmosis"
+	ProviderHuobi    ProviderName = "huobi"
+	ProviderOkx      ProviderName = "okx"
+	ProviderGate     ProviderName = "gate"
+	ProviderCoinbase ProviderName = "coinbase"
+	ProviderMock     ProviderName = "mock"
+)
+
+// ProviderName name of an oracle provider. Usually it is an exchange
+// but this can be any provider name that can give token prices
+// examples.: "binance", "osmosis", "kraken".
+type ProviderName string
```

Comment:
Might be a bit cleaner to add a .String() function and call it instead of using `string(p)` everywhere - wdyt?
```
func (p ProviderName) String() string {
	return string(p)
}
```

Reply:
I had done that way before - https://github.com/umee-network/umee/pull/1175#discussion_r931842536 
Robert commented that it would be more efficient `string(p)` 

In fact, it is, but it doesn't change that much

Reply:
In my opinion using `func (p ProviderName) String()` would be better for keeping methods in the same place the type is defined

Reply:
If we ever decide to change the `String()` method (make it all caps, etc) it would be cleaner to do so as well. Imo cleaner code > efficiency here, but either way doesn't change much atm.

Reply:
`string(p)` is a type conversion. `String` is a method. Two different things.

Reply:
> @RafilxTenfen then let's keep both the `Provider` type and the list in the existing `provider` package.
import cycle again... 
the good thing about being in the `types` package is that the `types` itself doesn't import any other package
@robert-zaremba